### PR TITLE
Add support for compilerOptions config property

### PR DESCRIPTION
**Pull request by chuckdumont**
This pull request adds support for setting Closure Compiler options in the Server-Side AMD config.  The options are specified by the <code>compilerOptions</code> config property. 
<p>Compiler config options are specified using [CompilerOptions](http://javadoc.closure-compiler.googlecode.com/git/com/google/javascript/jscomp/CompilerOptions.html) defined property name/value pairs. The names can be any property which is either declared as a public field or has a setter method which can be mapped to the property name using standard JavaBean property naming convention. The value is an array who's number and types of elements matches the number and types of the formal parameters defined by the setter method. If the setter method takes exactly one parameter, or the property is a public field which has no setter method, then the array notation may be dispensed with and the single parameter may be specified directly as in the following examples:
<p>
<pre>
compilerOptions: {
   defineToBooleanLiteral:['defineName', true]  // calls setDefineToBooleanLiteral("defineName", true);
}
compilerOptions: {
   acceptConstKeyword:true           // this is identical to acceptConstKeyword:[true].  Both
                                     // will invoke setAcceptConstKeyword(true);
}
compilerOptions: {
   aliasableStrings:[['foo', 'bar']] // setAliasableStrings takes a java.util.Set, so need to use a
                                     // nested array so that the string set will be passed as a single
                                     // parameter.
}
</pre>
<p>
The following JavaScript to Java property value conversions will be performed when attempting to match the provided values to the value types declared by the class:
<ul>
<li>Values specified as a JavaScript arrays will be converted to either a java.util.List or a java.util.Set as determined by the declared type of the receiving parameter.</li>
<li>If the declared type of the receiving parameter is an Enum and the provided value is a string which matches one of the Enum named constants, then the matching constant value will be used</li>
</ul>
<p>
<pre>
compilerOptions: {
   checkGlobalThisLevel:'WARNING'    // This invokes setCheckGlobalThisLevel(CheckLevel.WARNING);
}
</pre>

### CODE REVIEW

**Comment by chuckdumont** [comment]
@ddumont please review.  @cwgormanIBM FYI

File: jaggr-core/src/main/java/com/ibm/jaggr/core/util/CompilerUtil.java [hunk 1]
```diff
@@ -15,14 +15,400 @@
...
+	 * For example: <code>setXXX(arg1, ar2)</code>
+	 *
+	 * @param setterMethodName
+	 *            the method name
+	 * @param args
+	 *            the method arguments
+	 * @return the formatted string
+	 */
+	private static String formatForDisplay(String setterMethodName, List<Object> args) {
+		StringBuffer sb = new StringBuffer(CompilerOptions.class.getName());
```
Comment (ddumontatibm) [inline jaggr-core/src/main/java/com/ibm/jaggr/core/util/CompilerUtil.java]: You could write less code and use google's joiner

```java
Joiner.on(',').appendTo(new StringBuilder(), Lists.newArrayList());
```

**Comment by ddumontatibm** [comment]
otherwise lgtm

**Comment by chuckdumont** [comment]
Update PR to avoid use of CompilerOptions.clone() since it isn't implemented properly.  Instead, the JavaScriptModuleBuilder now calls CompilerUtil.compilerOptionsMapFromConfig() whenever the config is updated in order to obtain the config options as a map of java.lang.reflect.AccessibleObject/value pairs.  This map can be passed to CompilerUtil.applyCompilerOptionsMap() in order to efficiently apply the options to a new CompilerOptions object.  This is necessary because CompilerOptions provides no ability to clone an object or construct a new object from an existing object.

**Pull request merged**